src/components/LineChart/LineChart.props.ts

```diff
@@ -2,10 +2,12 @@ export interface IChartProps {
     title: string;
     width: number;
     height: number;
-    data: IChart[];
+    gridSize?: number;
+    data: IChartData[];
+    xAxisScale: string; // 'LINEAR' or 'TIME' 
 }
 
-export interface IChart {
-    time: Date;
+export interface IChartData {
+    key: Date | number;
```

Comment:
Not sure I think that it doesn't matter what data is called when we pass it in d3 we just need to specify what name of data is for what. I will change name to 'argument' it gives it a better meaning.